vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,570 @@
+
+# processed only head
+head(ctu05_data$baseline$hiv_date.datetime)
+
+# classes
+class(ctu05_data$baseline$visit_date)
+class(ctu05_data$baseline$visit_date.date)
+class(ctu05_data$baseline$hiv_date)
+class(ctu05_data$baseline$hiv_date.datetime)
 
-head(bmd_data[,c("patid", "visitdate", "bmd.bmd", "bmd.grouping", "bmd.age")])
-tail(bmd_data[,c("patid", "visitdate", "bmd.bmd", "bmd.grouping", "bmd.age")])
-table(bmd_data$bmd.grouping)
-unique(length(bmd_data$patid))
 ```
 
-As you can see there are 504 unique patients grouped into "Calcium" and "Placebo". The patient with
-id 999 was subsequently added to the original dataset. This record has some missing values.
-These values are missing on purpose, since missing values are needed to demonstrate some downstream 
-functionalities of the package. 
+Depending on the setup, incomplete dates can be valid entries in a secuTrial database.
+Thus they will also occasionally appear in your exports. The datetime conversion
+does not work in these cases and `NA`s are created. If this happens, `secuTrialR` will
+warn you accordingly and you should have a closer look into the affected
+datetime variables and whether you would like to perform so-called date imputation.
+
+## Export options
 
-The central files this vignette will be using are part of the package. Since the paths end up being a little
-cryptic we will store them in the following two variables. Both zip archives are secuTrial exports of the data
-above.
+The `secuTrialdata` object also contains information on the export options.
+\vspace{5pt}
 
 ```{r}
+ctu05_data$export_options
+```
+
+`export_options` itself is a `list`. If you are interested in more information
+than is printed you can also access it. Let's assume you would like to know
+the `project_name` and `encoding`.
+\vspace{5pt}
 
-export_location_shortnames <- system.file("extdata", "sT_exports", "BMD",
-                                         "s_export_CSV-xls_BMD_short_en_utf8.zip",
-                                          package = "secuTrialR")
+```{r}
+ctu05_data$export_options$project_name
+ctu05_data$export_options$encoding
+```
 
-export_location_longnames <- system.file("extdata", "sT_exports", "BMD",
-                                          "s_export_CSV-xls_BMD_long_en_utf8.zip",
-                                          package = "secuTrialR")
+Much more information is stored in the elements of `export_options`.
+The names of the elements should be descriptive enough to infer the contents.
+\vspace{5pt}
 
+```{r}
+names(ctu05_data$export_options)
 ```
 
-### Loading an export
+# Generic functions for `secuTrialdata` objects
+
+Now that you understand the `secuTrialdata` object we will show you some generic functions
+you can use on objects of this class.
 
-The code below shows how to load a secuTrial export. The output of read_secuTrial_raw is a list. 
+## Show the study participants
+
+First off you may be interested in a table of participants.
```

Comment:
Would it be helpful to have an explanation of "pat_id" at this point? E.g. by simply saying that it is derived from `mnpaid` and added to all form tables?